sbuild: add page (#12079)
tldr-pages/tldr: pages/common/sbuild.md

# sbuild

> > Build a Debian binary package in a clean `schroot`.

Reviewer: ```suggestion
> Build a Debian binary package in a clean `chroot` environment.
```
From the Debian wiki:
> [sbuild](https://tracker.debian.org/sbuild) is a convenience wrapper script of [schroot](https://tracker.debian.org/schroot) to build binary package easily under specified chroot.

> More information: <https://wiki.debian.org/sbuild>.

- Build the package in the current directory:

`sbuild`

- Build the given package:

`sbuild {{package}}`

- Build for a certain distribution:

> `sbuild --dist={{distribution}}`

Reviewer: ```suggestion
`sbuild --dist {{distribution}}`
```

- Build using custom dependencies:

> `sbuild --extra-package={{path/to/deb/files}}`

Reviewer: ```suggestion
`sbuild --extra-package {{path/to/deb_files}}`
```

> `sbuild --extra-package {{path/to/deb_files}}`

Reviewer 2: This sounds like the parameter should be a directory, but the documentation suggests otherwise:
> use the `--extra-package=./foo.deb` option

```suggestion
`sbuild --extra-package {{path/to/file.deb}}`
```

Author: It can be both in fact, and the example I would like to provide is passing a directory.

Author: --extra-package=package.deb|directory
              Make package.deb available for build-dependency resolution, by adding it to a temporary  archive  cre‐
              ated  by  sbuild.   This  makes  it easier to build packages against locally-built build dependencies,
              without waiting for those packages to enter the main archive, or going through the hassle of maintain‐
              ing a local archive and making it accessible inside the chroot.  package.deb is copied  into  the  ch‐
              root,  so  it can refer to any path on the host system.  If a directory is passed instead of a regular
              file, then all regular files inside that directory with a filename that ends in .deb will be added  in
              the  same fashion as it is done for individual packages.

> - Build using custom dependencies:
> 
> `sbuild --extra-package {{path/to/deb_files}}`

Reviewer 2: How about this?
```suggestion
- Build using custom dependencies (if a directory is passed, all files ending with `.deb` are used):

`sbuild --extra-package {{path/to/file_or_directory}}`
```

- Run a shell in case of build failure to further investigate:

> `sbuild --build-failed-commands=%SBUILD_SHELL`

Reviewer: ```suggestion
`sbuild --build-failed-commands %SBUILD_SHELL`
```

Reviewer 3: The docs seem to suggest using `=` for this command so IG we can leave this as it is and merge this PR.

- Cross build for a certain architecture:

> `sbuild --host={{architecture}}`

Reviewer: ```suggestion
`sbuild --host {{architecture}}`
```

- Build for the given native architecture:

> `sbuild --arch={{architecture}}`

Reviewer: ```suggestion
`sbuild --arch {{architecture}}`
```
